packages/flutter_driver/lib/src/common/find.dart

```diff
@@ -351,9 +351,9 @@ class Descendant extends SerializableFinder {
   /// Deserializes the finder from JSON generated by [serialize].
   static Descendant deserialize(Map<String, String> json) {
     final Map<String, String> jsonOfMatcher =
-        Map<String, String>.from(jsonDecode(json['of']));
+        Map<String, String>.from(jsonDecode(json['of']) as Map<String, dynamic>);
```

Comment:
This seems odd: we are casting to Map<String, dynamic>, but it actually seems to be a Map<String, String>?

Reply:
What about removing the double casts like this:

```dart
    dynamic value = json.decode('{"a": "b"}');
    print(value.cast<String, String>());
```

Reply:
It's also not possible to avoid a cast. The type of `value.cast<String, String>()` is `dynamic` as `value` is dynamic.

In the case of the commented line, you would have to use:

```dart
    final dynamic value = jsonDecode(json['of']);
    Map<String, String>.from(value.cast<String, String>() as Map<String, String>);
```

(In your sample you don't need a cast because `print` accepts `Object` as parameter)